### ISSUE TITLE
elgg_format_element() object to string conversion

### ISSUE THREAD

**Issue by hypeJunction** [Elgg/Elgg#10010]
Passing an array of objects to `elgg_format_element()` causes issues at:

```php
if (is_array($val)) {
   $val = implode(' ', $val);
}
```

**Comment by mrclay** [comment]
> Passing an array of objects

Why should the function support that?

**Comment by hypeJunction** [comment]
It shouldn't, but it tries to convert objects to strings.
We need to filter the array values before trying to concatenate them:

```php
elgg_format_attributes([
   'items' => [
     new ElggFile(),
     new ElggObject(),
   ]
]);
```

```php
if (is_array($val)) {
   $val = array_filter($val, function($e) {
      return is_scalara($e) || (is_object($e) && is_callable([$e, 'toString']));
   });
   $val = implode(' ', $val);
}

**Comment by mrclay** [comment]
I'd rather just throw if a non-scalar is present than to filter.

**Comment by hypeJunction** [comment]
Throwing will cause issues with `$vars`. We can just skip if array contains non-scalar values

**Comment by hypeJunction** [comment]
What if:

```php
elgg_format_attributes([
   'class' => [
      'elgg-foo',
      ['elgg-bar'],
      'elgg-baz',
   ],
]);

**Comment by mrclay** [comment]
😿 . This function can't be the Universal Serializer. Ignore non scalars and trigger an error?

**Comment by hypeJunction** [comment]
It triggers a php warning. I think we should just skip quietly. We inject
$vars a lot and that contains all kinds of data

**Comment by mrclay** [comment]
Fixed by #10013
